src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AvoidMultipleEnumerations/FlowAnalysis/GlobalFlowStateDictionaryFlowOperationVisitor.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.GlobalFlowStateAnalysis;
+
+namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines.AvoidMultipleEnumerations.FlowAnalysis
+{
+    using GlobalFlowStateDictionaryAnalysisData = DictionaryAnalysisData<AnalysisEntity, GlobalFlowStateDictionaryAnalysisValue>;
+    using GlobalFlowStateDictionaryAnalysisResult = DataFlowAnalysisResult<GlobalFlowStateDictionaryBlockAnalysisResult, GlobalFlowStateDictionaryAnalysisValue>;
+
+    internal abstract class GlobalFlowStateDictionaryFlowOperationVisitor : GlobalFlowStateDataFlowOperationVisitor<
+        GlobalFlowStateDictionaryAnalysisContext,
+        GlobalFlowStateDictionaryAnalysisResult,
+        GlobalFlowStateDictionaryAnalysisValue>
+    {
+        protected GlobalFlowStateDictionaryFlowOperationVisitor(
+            GlobalFlowStateDictionaryAnalysisContext analysisContext) : base(analysisContext, true)
+        {
+        }
+
+        protected override bool Equals(GlobalFlowStateDictionaryAnalysisData value1, GlobalFlowStateDictionaryAnalysisData value2)
+            => GlobalFlowStateDictionaryAnalysis.Domain.Equals(value1, value2);
+
+        protected override GlobalFlowStateDictionaryAnalysisValue GetAbstractDefaultValue(ITypeSymbol type)
+            => GlobalFlowStateDictionaryAnalysisValue.Empty;
+
+        protected override GlobalFlowStateDictionaryAnalysisData GetExitBlockOutputData(GlobalFlowStateDictionaryAnalysisResult analysisResult)
```

Comment:
This can't be moved to the base type because it is referencing GlobalFlowStateDictionaryAnalysisResult, 
which is `DataFlowAnalysisResult<GlobalFlowStateDictionaryBlockAnalysisResult, GlobalFlowStateDictionaryAnalysisValue>`, and this one is referencing this type `GlobalFlowStateDictionaryBlockAnalysisResult`